src/durable-cache/src/lib.rs

```diff
@@ -340,6 +338,46 @@ impl<C: DurableCacheCodec> DurableCache<C> {
 
         Ok(new_upper)
     }
+
+    /// Forcibly compacts the shard backing this cache in a background task. See
+    /// [`mz_persist_client::cli::admin::dangerous_force_compaction_and_break_pushdown`].
+    pub fn dangerous_compact_shard(
+        &self,
+        persist: PersistClient,
```

Comment:
Let's store the `PersistClient` we get in the constructor as a field on `DurableCache` so we don't have to re-plumb it back down here